cmd/crossplane/core/core.go

```diff
@@ -364,13 +364,25 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		return errors.Wrap(err, "cannot create client for API extension controllers")
 	}
 
+	// Create a separate no-cache client for use when the composite controller does not find an Unstructured
+	// resource that it expects to find in the cache.
+	nccl, err := client.New(mgr.GetConfig(), client.Options{
+		HTTPClient: mgr.GetHTTPClient(),
+		Scheme:     mgr.GetScheme(),
+		Mapper:     mgr.GetRESTMapper(),
+	})
```

Comment:
Does `client.New` definitely create an uncached client if you omit the `Cache` field from `client.Options`?

I'm sure it won't cache `Unstructured`. I'm worried there might be some default in the constructor that creates a cache for structured resources though. Not an issue in the context of this PR, but I wouldn't want someone to use this client for structured resources in future thinking it wasn't cached-backed if it actually was.

Reply:
I wondered the same thing and verified that it does what we want - https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/client.go#L180